spec/controllers/sufia/admin/admin_sets_controller_spec.rb

```diff
@@ -121,5 +121,31 @@
         expect(permission_template.workflow_name).to eq 'one_step_mediated_deposit'
       end
     end
+
+    describe "#destroy" do
+      let(:admin_set) { create(:admin_set, edit_users: [user]) }
+      context "with empty admin set" do
+        it "deletes the admin set" do
+          delete :destroy, params: { id: admin_set }
+          expect(response).to have_http_status(:found)
+          expect(response).to redirect_to(Sufia::Engine.routes.url_helpers.admin_admin_sets_path)
+          expect(flash[:notice]).to eq "Administrative set successfully deleted"
+        end
+      end
+      context "with a non-empty admin set" do
+        let(:work) { create(:generic_work, user: user) }
+        before do
+          admin_set.members << work
+          admin_set.reload
+        end
+        it "detaches the works and deletes the admin set" do
+          delete :destroy, params: { id: admin_set }
+          expect(response).to have_http_status(:found)
+          expect(response).to redirect_to(Sufia::Engine.routes.url_helpers.admin_admin_sets_path)
+          expect(flash[:notice]).to eq "Administrative set successfully deleted"
+          expect(GenericWork.exists?(work.id)).to be true
```

Comment:
Can you also verify here that the work no longer thinks it's in the deleted Admin Set?